server/plugin/hooks.go

```diff
@@ -26,13 +26,20 @@ func (p *Plugin) OnConfigurationChange() error {
 		return err
 	}
 
-	p.setConfiguration(configuration)
-
+	oldEncryptionSecret := p.getConfiguration().EncryptionSecret
 	mattermostSiteURL := p.API.GetConfig().ServiceSettings.SiteURL
 	if mattermostSiteURL == nil {
 		return errors.New("plugin requires Mattermost Site URL to be set")
 	}
 	configuration.MattermostSiteURL = *mattermostSiteURL
+	p.setConfiguration(configuration)
+
+	if oldEncryptionSecret != "" && oldEncryptionSecret != p.getConfiguration().EncryptionSecret {
+		if err := p.Store.DeleteUserOnSecretChange(); err != nil {
```

Comment:
we are deling whole user or just its token ?

Reply:
we are deleting details related to OAuth, like access token, refresh token

Reply:
then DeleteUserTokenOnSecretChange is a better name for that function then